### PR TITLE
add support for rabbitmq

### PR DESCRIPTION
**Pull request by jiang4869**
`listener.go` 实现了监听rabbitmq的功能
`sender.go` 实现了rabbitmq消息发送的功能
`rabbitmqadmin.go` 实现了rabbitmq交换机的创建、消息队列的创建、交换机和消息队列绑定的功能
`example/rabbitmq/` 使用的例子

### CODE REVIEW

File: rabbitmq/listener.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		q.ErrorHandler(err, "failed to listener")
+		go func() {
+			for d := range msg {
+				if err := q.handler.Consume(string(d.Body)); err != nil {
+					logx.Errorf("Error on consuming: %s, error: %v", string(d.Body), err)
+				}
+			}
+		}()
+	}
+	<-q.forever
```
Comment (kevwan) [inline rabbitmq/listener.go:76]: What's this for?

Comment (jiang4869) [reply rabbitmq/listener.go:76]: > What's this for?

用于阻塞线程。没加的话，启动后程序就直接停止了

File: rabbitmq/config.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ type ConsumerConf struct {
 
 type RabbitMqSenderConf struct {
 	RabbitMqConf
+	ContentType string
```
Comment (kevwan) [inline rabbitmq/config.go:28]: Any default value for `ContentType`?

Comment (jiang4869) [reply rabbitmq/config.go]: > Any default value for `ContentType`?

MIME content type

File: rabbitmq/config.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	Name      string
+	AutoAck   bool `json:",default=true"`
+	Exclusive bool `json:",default=false"`
+	NoLocal   bool `json:",default=false"` // Set to true, which means that messages sent by producers in the same connection cannot be delivered to consumers in this connection
+	NoWait    bool `json:",default=false"` // Whether to block processing
+}
+
+type RabbitMqSenderConf struct {
+	RabbitMqConf
+	ContentType string // MIME content type
```
Comment (kevwan) [inline rabbitmq/config.go:28]: Can we put a default value here?

Comment (jiang4869) [reply rabbitmq/config.go:28]: > Can we put a default value here?

You can set the default value `text/plain`。Other optional values, such as `application/json`, `application/xml` and other MIME types

Comment (kevwan) [reply rabbitmq/config.go:28]: I mean:

ContentType string `json:",default=text/plain"`

Comment (jiang4869) [reply rabbitmq/config.go]: > I mean:
> 
> ContentType string `json:",default=text/plain"`

ok

**Pull request merged**